### ISSUE TITLE
Implement Feature Adversaries attack framework-specific in PyTorch and TensorFlow

### ISSUE THREAD

**Issue by beat-buesser** [Trusted-AI/adversarial-robustness-toolbox#949]
The Feature Adversaries attack is an important attack for robustness evaluations and is available in ART in a framework-agnostic implementation in `art.attacks.evasion.FeatureAdversaries`. Because of its importance we would like to propose the implementation for framework-specific version in TensorFlow and PyTorch to take advantage of improved performance provided by these frameworks.

Paper link: https://arxiv.org/abs/1511.05122
